### PR TITLE
[proxima-beam-core] register common types in Beam Pipeline

### PR DESCRIPTION
**Pull request by je-ik**

### CODE REVIEW

**Comment by je-ik** [comment]
cc @dmvk

File: beam/core/src/main/java/cz/o2/proxima/beam/core/BeamDataOperator.java [hunk 1]
```diff
@@ -471,6 +481,23 @@ public boolean hasDirect() {
...
+    if (!typesRegisteredFor(ret.getPipeline())) {
+      registerTypesFor(ret.getPipeline());
+    }
     return ret;
   }
+
+  private void registerTypesFor(Pipeline pipeline) {
+    CoderRegistry registry = pipeline.getCoderRegistry();
+    registry.registerCoderForClass(GlobalWindow.class, GlobalWindow.Coder.INSTANCE);
+    registry.registerCoderForClass(IntervalWindow.class, IntervalWindowCoder.of());
```
Comment (dmvk) [inline beam/core/src/main/java/cz/o2/proxima/beam/core/BeamDataOperator.java:493]: do we really need to register coders for beam windows? aren't they already registred?

File: beam/core/src/main/java/cz/o2/proxima/beam/core/BeamDataOperator.java [hunk 2]
```diff
@@ -471,6 +481,23 @@ public boolean hasDirect() {
...
+    registry.registerCoderForClass(StreamElement.class, StreamElementCoder.of(repo));
+    registry.registerCoderForClass(EntityDescriptor.class, EntityDescriptorCoder.of(repo));
+    registry.registerCoderForClass(AttributeDescriptor.class, AttributeDescriptorCoder.of(repo));
+    typesRegistered.put(pipeline, true);
+  }
+
+  private boolean typesRegisteredFor(Pipeline pipeline) {
+    return Optional.ofNullable(typesRegistered.get(pipeline)).orElse(false);
```
Comment (dmvk) [inline beam/core/src/main/java/cz/o2/proxima/beam/core/BeamDataOperator.java:501]: may you could just turn 'typesRegistered' into a `Set` an check for element presence over here? I'd say this is just kind of complicated presence check...

**Comment by dmvk** [comment]
can you please check failing travis build?

File: beam/core/src/main/java/cz/o2/proxima/beam/core/BeamDataOperator.java (hunk 1 above)
Comment (je-ik) [reply beam/core/src/main/java/cz/o2/proxima/beam/core/BeamDataOperator.java:493]: unfortunately, they aren't.

File: beam/core/src/main/java/cz/o2/proxima/beam/core/BeamDataOperator.java (hunk 2 above)
Comment (je-ik) [reply beam/core/src/main/java/cz/o2/proxima/beam/core/BeamDataOperator.java:501]: emmm .... :)

Comment (je-ik) [reply beam/core/src/main/java/cz/o2/proxima/beam/core/BeamDataOperator.java:501]: fixed

**Pull request merged**